### ISSUE TITLE
UX Improvements for Schedulers

### ISSUE THREAD

**Issue by ravi-mosaicml** [mosaicml/composer#591]
Could we convert our schedulers into classes, where ssr and state are passed in on __call__, and all other (scheduler-specific) parameters are passed in on init. It's not a great UX that users will have to manually construct lambas to use our schedulers.

See the documentation page here: https://mosaicml-composer--531.com.readthedocs.build/en/531/trainer/optimizers_and_schedulers.html#composerschedulers

Instead, I think it would be cleaner if the user could pass in a `StepScheduler(step_size=['5ep', '25ep'])`, and then the trainer calls `compile_scheduler(step_scheduler, state, ssr)`.

This would also solve this comment which was originally on #463:

To simplify the logic in the training loop and to remove the `functools.partial` in scale scheduler, could `compile_scheduler` take the `ssr` as a parameter?
Then on line 63, we can call the custom scale_pytorch_scheduler logic, and on line 72, we can pass the ssr into the `return scheduler(state, ssr)`

Just not a big fan of functools.partial as they are not typechecked.

_Originally posted by @ravi-mosaicml in https://github.com/mosaicml/composer/pull/463#discussion_r810257320_

**Comment by growlix** [comment]
Came here to file this same issue after updating a tutorial notebook, where I instantiate the scheduler by calling the `SchedulerHParams` object then calling `initialize_object()`, which feels clunky. +1 to your solution, Ravi!

**Comment by dblalock** [comment]
+1 for having classes rather than requiring `functools.partial`. Much more discoverable in the docs + consistent with normal torch schedulers. Also would be nice to support (or auto-convert? or wrap?) vanilla torch schedulers, but maybe that's not feasible.

**Comment by hanlint** [comment]
Could you clarify what you mean by wrap? You should be able to supply torch schedulers directly via:
```
from torch.optim.lr_scheduler import StepLR

trainer = Trainer(schedulers=StepLR(..))
```

**Comment by ravi-mosaicml** [comment]
> discoverable

Vanilla torch schedulers are supported as well; it's buried in the scheduler type definition.
